lib/fastlane/plugin/wpmreleasetoolkit/models/firebase_test_runner.rb

```diff
@@ -20,20 +20,21 @@ def self.preflight(verify_gcloud_binary: true, verify_logged_in: true)
     # @param [FirebaseDevice] device The virtual device to run tests on.
     # @param [String] type The type of test to run.
     #
-    def self.run_tests(project_id:, apk_path:, test_apk_path:, device:, type: 'instrumentation')
+    def self.run_tests(project_id:, apk_path:, test_apk_path:, device:, test_targets: nil, type: 'instrumentation')
       raise "Unable to find apk: #{apk_path}" unless File.file?(apk_path)
       raise "Unable to find apk: #{test_apk_path}" unless File.file?(test_apk_path)
       raise "Invalid Type: #{type}" unless VALID_TEST_TYPES.include?(type)
 
-      command = Shellwords.join [
-        'gcloud', 'firebase', 'test', 'android', 'run',
-        '--project', project_id,
-        '--type', type,
-        '--app', apk_path,
-        '--test', test_apk_path,
-        '--device', device.to_s,
-        '--verbosity', 'info',
-      ]
+      params = {
+        project: project_id,
+        type: type,
+        app: apk_path,
+        test: test_apk_path,
+        'test-targets': test_targets,
+        device: device.to_s,
+        verbosity: 'info'
+      }.compact.flat_map { |k, v| ["--#{k}", v] }
+      command = Shellwords.join(['gcloud', 'firebase', 'test', 'android', 'run', *params])
```

Comment:
@AliSoftware Can you explain the benefits of switching to this syntax from the previous syntax? I'm not questioning it, just curious 😃

Reply:
This allowed me to inject the `compact` on the `Hash` before transforming it into an array of key-value-pairs.

The effect of that `.compact` is to remove any entry from the `Hash` with a `nil` value, which was necessary to completely remove `--test-targets <value>` from the params if the user didn't provide any value for it. Otherwise we'd be passing `--test-targets ""` to the `gcloud` command instead of skipping that param entirely when it's not provided by the user.

Reply:
Makes sense, thank you for explaining!